main/acle.md

```diff
@@ -1072,6 +1073,30 @@ including the header:
 
 Including `<arm_sme.h>` also includes [`<arm_sve.h>`](#arm_sve.h).
 
+### Predefined feature macros and header files
+
+Testing a feature macro returns the compiler's availability for the feature
+via ACLE documented intrinsics and inline assembly. This property can be
+independent of where the test is performed. There should be no assumption on
+the order or context in wich the preprocessor macros are evaluated.
+
+The compiler may add additional restrictions to the intrinsics beyond what is
+captured by the ACLE macros depending on the context in which the intrinsics
+are used. For example:
+
+``` c
+    #include <arm_sme.h>
+    void foo(svbool_t pg, void *ptr, uint32_t slice_base) {
+    #ifdef __ARM_FEATURE_SME
+      svst1_hor_za8(0, slice_base, pg, ptr);
+    #endif
+    }
+```
+
+While testing `__ARM_FEATURE_SME` ensures that the SME intrinsic `svst1_hor_za8`
+is available, `foo` will fail to compile because the call does not occur in a
+[streaming statement](#streaming-statement).
```

Comment:
```suggestion
If `__ARM_FEATURE_SME` evaluates to `true` the SME intrinsic `svst1_hor_za8`
is available, but `foo` may still fail to compile because the call does not occur in a
[streaming statement](#streaming-statement).
```